refactor: adopt compacted Core version naming (V3_110)

Unisay opened the pull request:
## Summary

Aligns dropList naming with array builtin convention from PR #12.

## Changes

**Module renames:**
- `V_1_0.hs` → `V3_100.hs`
- `V_1_1.hs` → `V3_110.hs`

**Identifier updates:**
- `succeedingDropListPolicyCompiledV3` → `succeedingDropListPolicyCompiled_V3_110`
- `expensiveDropListScriptGroupV3` → `expensiveDropListScriptGroup_V3_110`

**Script output names:**
- `succeedingDropListPolicyScriptV3.plutus` → `succeedingDropListPolicyScript_V3_110.plutus`
- `expensiveDropListPolicyScriptV3_N.plutus` → `expensiveDropListPolicyScript_V3_110_N.plutus`

**Qualified imports:**
- `Batch6_1_1` → `Batch6_V3_110`

**Added CLAUDE.md documentation** of naming convention.

## Convention

Uses compacted Plutus Core digits for clearer separation:
- **Underscore** separates distinct concepts (Plutus version from Core version)
- **No separator** within Core version digits (compacted: 110, 100)

**Examples:**
- `V3_110` = PlutusV3 + Plutus Core 1.1.0
- `V3_100` = PlutusV3 + Plutus Core 1.0.0

**Benefits:**
- Less verbose: `V3_110` vs `V_1_1` (clearer hierarchy)
- Visual clarity: immediately see what each underscore separates
- Consistent with arrays: same pattern across all Batch6 builtins
- Better grepping: `V3_*` finds all V3, `*_110` finds all 1.1.0

## Generated Scripts (6 total)

- `succeedingDropListPolicyScript_V3_110.plutus` (642 bytes)
- `expensiveDropListPolicyScript_V3_110_1.plutus` (535 bytes)
- `expensiveDropListPolicyScript_V3_110_2.plutus` (535 bytes)
- `expensiveDropListPolicyScript_V3_110_3.plutus` (1.3K)
- `expensiveDropListPolicyScript_V3_110_4.plutus` (1.3K)
- `expensiveDropListPolicyScript_V3_110_5.plutus` (1.3K)

## Testing

- [x] Build succeeds with `-Werror`
- [x] All 6 scripts regenerate correctly
- [x] Naming consistent with array builtins

## Related

- PR #12: Array builtins (establishes convention)
- Issue #14: Full version matrix for array builtins
- This PR: Applies same convention to dropList for consistency

Unisay committed:
refactor: adopt compacted Core version naming (V3_110)

Align dropList with array builtin naming convention:
- Old: V_1_1 → New: V3_110
- Old: V_1_0 → New: V3_100

Uses compacted Plutus Core version digits for clarity:
- Underscore: separates Plutus version from Core version
- No separator: within Core version digits (110, 100)

Changes:
- Rename modules: V_X_Y.hs → V3_{XY0}.hs
- Update identifiers: ...V3 → ..._V3_110
- Update script outputs: ...V3.plutus → ..._V3_110.plutus
- Qualified imports: Batch6_V3_110
- Add CLAUDE.md with naming convention documentation

Generated scripts (6 total):
- succeedingDropListPolicyScript_V3_110.plutus
- expensiveDropListPolicyScript_V3_110_{1-5}.plutus

Benefits:
- Less verbose: V3_110 vs V_1_1 (clearer separation)
- Consistent with array builtins convention (PR #12)
- Clear hierarchy: version concepts (separated) vs digits (compact)

Convention documented in CLAUDE.md for project-wide consistency.

Related: PR #12 (array builtins), issue #14

kwxm reviewed (changes requested):
Thanks for fixing this.  We have to add tests for quite a few new builtins, so I think we should makea `Batch6` directory and add all of the new things there so that we can keep track of them (and perhaps later (but not right now!) go back and do this for at least the Batch 5 scripts as well).

Unisay committed:
refactor: reorganize dropList into Batch6/DropList/ subdirectory

Move dropList to subdirectory structure for consistency with batch6
organization pattern, following Kenneth's feedback.

Directory structure:
  Batch6/
  └── DropList/
      ├── Common.hs (was DropList.hs)
      ├── V3_100.hs
      └── V3_110.hs

Changes:
- DropList.hs → DropList/Common.hs
- V3_100.hs → DropList/V3_100.hs
- V3_110.hs → DropList/V3_110.hs
- Update module paths: Batch6.DropList.V3_110
- Update imports: DropList_V3_110
- Update cabal exposed-modules

This structure enables clean organization of all batch6 builtins
(arrays, dropList, etc.) with each builtin in its own subdirectory.

Generated scripts (6 total):
- succeedingDropListPolicyScript_V3_110.plutus
- expensiveDropListPolicyScript_V3_110_{1-5}.plutus

Related: Kenneth's feedback on PR #15, PR #12 (arrays)

Unisay committed:
docs: update CLAUDE.md examples for Batch6 subdirectory structure

Update examples to reflect reorganization:
- PlutusScripts.Batch6.Array.V3_110
- PlutusScripts.Batch6.DropList.V3_110

Matches PR #12 documentation updates.

Unisay committed:
refactor: simplify identifiers by removing version repetition

Remove version suffix from identifiers to avoid repetition:
- Old: DropList_V3_110.succeedingDropListPolicyCompiled_V3_110
- New: DropList_V3_110.succeedingDropListPolicy

Changes:
- Remove version suffix from all exported identifiers
- Version info already clear from qualified import

Example usage:
  DropList_V3_110.succeedingDropListPolicy
  DropList_V3_110.expensiveDropListScriptGroup

Benefits: cleaner code, no excessive repetition, version still clear.
Matches array builtin pattern from PR #12.

Unisay committed:
refactor: introduce VersionedScript pattern for batch6 builtins

Apply same pattern as array builtins (PR #12) to reduce boilerplate.

New pattern:
- Added Helpers/Envelopes with VersionedScript types
- Created PlutusScripts/Batch6/DropList overarching module
- Exports allDropListScripts and allDropListScriptGroups lists
- Main.hs: mapM_ writeVersionedScript/writeVersionedScriptGroup

Benefits:
- Main.hs scales: 2 lines for dropList (not 7 lines)
- No version repetition: DropList_V3_110.succeedingDropListPolicy
- Type-safe version tracking with proper PlutusCore type
- Matches array builtin pattern for consistency

Example:
  Old:
    writeEnvelopeV3 "succeedingDropListPolicyScript_V3_110"
      DropList_V3_110.succeedingDropListPolicy
    writeScriptGroup DropList_V3_110.expensiveDropListScriptGroup

  New:
    mapM_ writeVersionedScript DropList.allDropListScripts
    mapM_ writeVersionedScriptGroup DropList.allDropListScriptGroups

Generated scripts (6 total):
- succeedingDropListPolicyScript_V3_110.plutus
- expensiveDropListPolicyScript_V3_110_{1-5}.plutus

Related: PR #12 (array pattern), Kenneth's feedback

kwxm reviewed (approved):
Thanks for the updates.
